core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -228,4 +229,35 @@ private TableProperties() {
 
   public static final String MERGE_CARDINALITY_CHECK_ENABLED = "write.merge.cardinality-check.enabled";
   public static final boolean MERGE_CARDINALITY_CHECK_ENABLED_DEFAULT = true;
+
+  /**
+   * Encryption manager type
+   */
+  public static final String ENCRYPTION_MANAGER_TYPE = "encryption.manager.type";
+  public static final String ENCRYPTION_MANAGER_TYPE_PLAINTEXT = "plaintext";
+  public static final String ENCRYPTION_MANAGER_TYPE_SINGLE_ENVELOPE = "envelope";
```

Comment:
Why not using "single.envelop"?

Reply:
The standard technique is called "envelope encryption", so "envelope" is more intuitive / better known.